src/test/codegen/swap-small-types.rs

```diff
@@ -1,6 +1,7 @@
 // compile-flags: -O
 // only-x86_64
 // ignore-debug: the debug assertions get in the way
+// ignore-test for now (this is just to get CI happy)
```

Comment:
You probably figured this out already, but as the person who added a bunch of these codegen tests, the `i48` part is usually incidental, just there as the easiest way to accurately check what it wanted to confirm.  If a different encoding of the rust types into llvm types is better, I'm all for it.

Reply:
Yeah, I did see that and what this PR did was to always use the same type (no more aggregation like `[f32; 4]` -> `u128`) so that using them in/with a function didn't require extra steps witch works well for generating better assembly but at the cost of having way to much perf regression.

If you have a idea to reduce them, I'm all in.

Reply:
I must also say that the transformation was only done to small types (max 128 bits in x86_64), this PR wouldn't have changed anything for them.